src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/SimpleStrandSwitchVariantDetector.java

```diff
@@ -0,0 +1,301 @@
+package org.broadinstitute.hellbender.tools.spark.sv.discovery.prototype;
+
+import avro.shaded.com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.CigarOperator;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.apache.logging.log4j.Logger;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.StructuralVariationDiscoveryArgumentCollection;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.*;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.RDDUtils;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.SVVCFWriter;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.SerializablePredicate;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.SvCigarUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import scala.Tuple2;
+import scala.Tuple3;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+final class SimpleStrandSwitchVariantDetector implements VariantDetectorFromLocalAssemblyContigAlignments {
+
+    @SuppressWarnings("unchecked")
+    private static final List<String> EMPTY_INSERTION_MAPPINGS = Collections.EMPTY_LIST;
+
+    static final int MORE_RELAXED_ALIGNMENT_MIN_LENGTH = 30;
+    static final int MORE_RELAXED_ALIGNMENT_MIN_MQ = 20;
+
+    public void inferSvAndWriteVCF(final JavaRDD<AlignedContig> longReads, final String vcfOutputFileName,
+                                   final Broadcast<ReferenceMultiSource> broadcastReference, final String fastaReference,
+                                   final Logger toolLogger) {
+
+        longReads.cache();
+        toolLogger.info(longReads.count() + " chimera indicating either 1) simple strand-switch breakpoints, or 2) inverted duplication.");
+
+        // split between suspected inv dup VS strand-switch breakpoint
+        final Tuple2<JavaRDD<AlignedContig>, JavaRDD<AlignedContig>> split
+                = RDDUtils.split(longReads.map(SimpleStrandSwitchVariantDetector::removeOverlap),
+                                  new IsLikelyInvertedDuplication(), true);
+
+        final JavaRDD<VariantContext> simpleStrandSwitchBkpts =
+                dealWithSimpleStrandSwitchBkpts(split._2, broadcastReference, toolLogger);
+        SVVCFWriter.writeVCF(null, vcfOutputFileName.replace(".vcf", "_simpleSS.vcf"),
+                fastaReference, simpleStrandSwitchBkpts, toolLogger);
+
+        // TODO: 8/23/17 add inv dup code in the next pr
+    }
+
+    public static final class IsLikelyInvertedDuplication implements SerializablePredicate<AlignedContig> {
```

Comment:
Your SerializablePredicate is semi-redundant with org.apache.spark.api.java.function.Function<T, Boolean>, which is what JavaRDD::filter expects.  You could eliminate it, and just use the spark Function interface in RDDUtils:
```
import org.apache.spark.api.java.function.Function;
...
    public static <T> Tuple2<JavaRDD<T>, JavaRDD<T>> split(
            final JavaRDD<T> input,
            final Function<T, Boolean> predicate,
            final boolean unpersistInputRDD) {

        final JavaRDD<T> first = input.filter(predicate).cache();
        final JavaRDD<T> second = input.filter(t -> !predicate.call(t)).cache();

        if (unpersistInputRDD) input.unpersist();
        return new Tuple2<>(first, second);
    }
```

And then in this simple circumstance you can eliminate your explicit static class in favor of a lambda.  The call at line 53 above becomes:
```
        final Tuple2<JavaRDD<AlignedContig>, JavaRDD<AlignedContig>> split
                = RDDUtils.split(contigs.map(SimpleStrandSwitchVariantDetector::removeOverlap),
                                  contig ->
                                          BreakpointComplications.isLikelyInvertedDuplication(
                                                  contig.alignmentIntervals.get(0),
                                                    contig.alignmentIntervals.get(1)),
                                  true);
```

This doesn't prevent you from implementing Function in an explicit static class when it's complicated enough to warrant it.

Does that help?

Reply:
By the way, I'm a little skeptical of putting the caching behavior into RDDUtils.split.
I fear it might not always be appropriate (if the RDD is very large, for example), but a client might forget that the caching is happening, and might easily forget to unpersist the split RDDs.
Better, IMO, to have the cache and unpersist be called in the same code block, so that it's absolutely obvious how executor memory is being managed.

Reply:
Thanks for the suggestion, @tedsharpe. For some unknown reason, the lambda lead to a task not serializable exception when I first attempted as in your suggestion, and that is why I had this Functor wrapper around a function. I'll try again and see if it works with a simple lambda.

Reply:
For caching the split-up two RDDs, I agree and the caching will be removed in the next commit.

Reply:
The trick is that the spark functional interfaces are all marked Serializable.  So you have to use one of those, rather than, for example, Java's Predicate interface, which isn't Serializable out of the box.

Reply:
Right, that was why I had this trivial SerializablePredicate interface, without knowing that there is this `Function` class in Spark as well.

Reply:
It would have been nice if Spark had extended Predicate with serializability, rather than reusing Function with a Boolean return type for its filter method.  That way the negative test could have used predicate.negate().  Oh, well.